main_2024.py

```diff
@@ -81,14 +84,24 @@ def main() -> int:
         # pylint: disable=invalid-name
         QUEUE_MAX_SIZE = config["queue_max_size"]
 
-        VIDEO_INPUT_CAMERA_NAME = config["video_input"]["camera_name"]
         VIDEO_INPUT_WORKER_PERIOD = config["video_input"]["worker_period"]
-        VIDEO_INPUT_SAVE_NAME_PREFIX = config["video_input"]["save_prefix"]
-        VIDEO_INPUT_SAVE_PREFIX = str(pathlib.Path(logging_path, VIDEO_INPUT_SAVE_NAME_PREFIX))
+        VIDEO_INPUT_OPTION = camera_factory.CameraOption(config["video_input"]["camera_option"])
+        VIDEO_INPUT_WIDTH = config["video_input"]["width"]
+        VIDEO_INPUT_HEIGHT = config["video_input"]["height"]
+        if VIDEO_INPUT_OPTION == camera_factory.CameraOption.OPENCV:
+            VIDEO_INPUT_CAMERA_CONFIG = camera_opencv.ConfigOpenCV(
+                **config["video_input"]["camera_config"]
+            )
+        elif VIDEO_INPUT_OPTION == camera_factory.CameraOption.PICAM2:
+            VIDEO_INPUT_CAMERA_CONFIG = camera_picamera2.ConfigPiCamera2(
+                **config["video_input"]["camera_config"]
+            )
```

Comment:
Might be worth putting these options into separate config files. For example:
`config.yaml`
```yaml
camera_enum: 0
# Other common camera options
camera_configs:
- camera_opencv.yaml
- camera_picam2.yaml
```
`main.py`
```py
camera_option = config["camera_enum"]
camera_config_path = config["camera_configs"][camera_option.value]
...

# Camera specific configuration
... = read_yaml.open_config(camera_config_path)
```

Reply:
I feel like it is easier to keep it in one config file. It's easier to find, and these options aren't that big either.